.github/workflows/tests.yaml

```diff
@@ -140,6 +140,7 @@ jobs:
     needs: [rubocop, skip_duplicate_runs]
     env:
       CI: true
+      PUMA_TEST_DEBUG: true
```

Comment:
Do we want to leave this on?

Reply:
I suppose there's no reason it can't be on for people running tests locally.

IIRC, Minitest runs our code that logs the debugging info in a 'Kernel.at_exit' hook.  Puma::ControlCLI has several `exit` statements, which can cause interesting results in tests if they're not trapped.

Also, the Actions logging syntax lets us collapse the section, that won't work locally in a terminal/console window.

Finally, I added it a while ago because I wanted to log some metrics from tests (passing or not), and I didn't want the info scattered throughout the test logs.  The CI systems are a bit resource limited compared to many devs' local PC's...